packages/core/src/util/token.ts

```diff
@@ -26,20 +26,20 @@ import { jwtVerify, createRemoteJWKSet } from "jose";
 type WithStack = { stack: string };
 
 /**
- * Extract a WebID from an ID token payload based on https://github.com/solid/webid-oidc-spec.
+ * Extract a WebID and the clientID from an ID token payload based on https://github.com/solid/webid-oidc-spec.
  * Note that this does not yet implement the user endpoint lookup, and only checks
- * for `webid` or IRI-like `sub` claims.
+ * for `webid`, `azp` or IRI-like `sub` claims.
  *
  * @param idToken the payload of the ID token from which the WebID can be extracted.
- * @returns a WebID extracted from the ID token.
+ * @returns an object with entries webId and clientId extracted from the ID token.
  * @internal
  */
 export async function getWebidFromTokenPayload(
   idToken: string,
   jwksIri: string,
   issuerIri: string,
   clientId: string,
-): Promise<string> {
+): Promise<{ webId: string; clientId: string }> {
   let payload: JWTPayload;
```

Comment:
```suggestion
  let payload: JWTPayload;
  let webId: string;
  let clientId: string; 
```

Reply:
I may need to initialise the `clientId` to any value, like `""`, for it not to complain about returning a variable before it being assigned. Then, return undefined or its value depending if it is `""` or not.
Does it sound reasonable?

Reply:
let's do it the other way, and initialize it as undefined, and set it to a `string` if appropriate (adjusting the declaration to `let clientId: string | undefined = undefined`. If I remember correctly, this should be equivalent to the current `let clientId: string;`, because variables are initialized to `undefined` by default, but I'm not 100% sure of that.